homeassistant/components/ebus/ebusd.py

```diff
@@ -0,0 +1,189 @@
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Ebusd..."""
+    name = config.get(CONF_NAME)
+    circuit = config.get(CONF_CIRCUIT)
+    server_address = (config.get(CONF_HOST), config.get(CONF_PORT))
+
+    try:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        data = EbusdData(server_address, circuit)
+
+        sock.settimeout(5)
+        sock.connect(server_address)
+        sock.close()
+
+        dev = []
+        for variable in config[CONF_MONITORED_CONDITIONS]:
+            dev.append(Ebusd(data, circuit, variable, name))
+
+        add_devices(dev)
+        hass.services.register(DOMAIN, SERVICE_EBUSD_WRITE, data.write)
+    except socket.timeout:
+        raise PlatformNotReady
+    except socket.error:
+        raise PlatformNotReady
+
+
+def timer_format(string):
+    """Datetime formatter."""
+    _r = []
+    _s = string.split(';')
+    for i in range(0, len(_s) // 2):
+        if(_s[i * 2] != '-:-' and _s[i * 2] != _s[(i * 2) + 1]):
+            _r.append(_s[i * 2] + '/' + _s[(i * 2) + 1])
+    return ' - '.join(_r)
+
+
+class EbusdData:
+    """Get the latest data from Ebusd."""
+
+    def __init__(self, address, circuit):
+        """Initialize the data object."""
+        self._circuit = circuit
+        self._address = address
+        self.value = {}
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self, name):
+        """Call the Ebusd API to update the data."""
+        import ebusdpy
+        command = READ_COMMAND.format(self._circuit, name, CACHE_TTL)
+
+        try:
+            _LOGGER.debug("Opening socket to ebusd %s: %s", name, command)
+            command_result = ebusdpy.send_command(self._address, command)
+            if 'not found' in command_result:
+                _LOGGER.warning("Element not found: %s", name)
+                raise RuntimeError("Element not found")
+            else:
+                self.value[name] = command_result
+        except socket.timeout:
+            _LOGGER.error("socket timeout error")
+            raise RuntimeError("socket timeout")
+        except socket.error:
+            _LOGGER.error("socket error: %s", socket.error)
+            raise RuntimeError("Command failed")
+
+    def write(self, call):
+        """Call write methon on ebusd."""
+        import ebusdpy
+        name = call.data.get('name')
+        value = call.data.get('value')
+        command = WRITE_COMMAND.format(self._circuit, name, value)
+
+        try:
+            _LOGGER.debug("Opening socket to ebusd %s: %s", name, command)
+            command_result = ebusdpy.send_command(self._address, command)
+            if 'done' not in command_result:
+                _LOGGER.warning('Write command failed: %s', name)
+        except socket.timeout:
+            _LOGGER.error("socket timeout error")
+        except socket.error:
+            _LOGGER.error()
+
+
+class Ebusd(Entity):
```

Comment:
Why are we adding a sensor entity to a ebus component platform? Shouldn't this be a sensor component platform?

Reply:
I believe that the author tried to pull this as a sensor before, but @balloob has rejected that PR asking CrazYoshi to rewrite this as a component. 
Also there's a huge potential for this component to grow in the future, right now it's mostly reading sensor values, but in the future it may also get binary sensors, switches, or even thermostat functionality.

Reply:
Yes exactly, if you scroll up you can see the change requested by him. That's the reason why I start a new component. Ebus is a generic container where also other component using the same standard can be put inside.

Reply:
So what should I do to this component again? I didn't get
Can you give me some suggestions?

Reply:
Component manages connection and hosts services, sensor platform of ebusd will show some info as a sensor entity.